src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -126,6 +146,95 @@ void IslandWindow::_HandleCreateWindow(const WPARAM, const LPARAM lParam) noexce
     UpdateWindow(_window.get());
 }
 
+// Method Description:
+// - Handles a WM_SIZING message, which occurs when user drags a window border
+//   or corner. It intercepts this resize action and applies 'snapping' i.e.
+//   aligns the terminal's size to its cell grid. We're given the window size,
+//   which we then adjust based on the terminal's properties (like font size).
+// Arguments:
+// - wParam: Specifies which edge of the window is being dragged.
+// - lParam: Pointer to the requested window rectangle (this is, the one that
+//   originates from current drag action). It also acts as the return value
+//   (it's a ref parameter).
+// Return Value:
+// - <none>
+LRESULT IslandWindow::_OnSizing(const WPARAM wParam, const LPARAM lParam)
+{
+    if (!_pfnSnapDimensionCallback)
+    {
+        // If we haven't been given the callback that would adjust the dimension,
+        // then we can't do anything, so just bail out.
+        return FALSE;
+    }
+
+    LPRECT winRect = reinterpret_cast<LPRECT>(lParam);
+
+    // Find nearest monitor.
+    HMONITOR hmon = MonitorFromRect(winRect, MONITOR_DEFAULTTONEAREST);
+
+    // This API guarantees that dpix and dpiy will be equal, but neither is an
+    // optional parameter so give two UINTs.
+    UINT dpix = USER_DEFAULT_SCREEN_DPI;
+    UINT dpiy = USER_DEFAULT_SCREEN_DPI;
+    // If this fails, we'll use the default of 96.
+    GetDpiForMonitor(hmon, MDT_EFFECTIVE_DPI, &dpix, &dpiy);
+
+    const auto nonClientSize = GetTotalNonClientExclusiveSize(dpix);
+    auto clientWidth = winRect->right - winRect->left - nonClientSize.cx;
+    auto clientHeight = winRect->bottom - winRect->top - nonClientSize.cy;
+
+    if (wParam != WMSZ_TOP && wParam != WMSZ_BOTTOM)
+    {
+        // If user has dragged anything but the top or bottom border (so e.g. left border,
+        // top-right corner etc.), then this means that the width has changed. We thus ask to
+        // adjust this new width so that terminal(s) is/are aligned to their character grid(s).
+        clientWidth = static_cast<int>(_pfnSnapDimensionCallback(true, static_cast<float>(clientWidth)));

```

Comment:
prefer gsl::narrow or gsl::narrow_cast to static_cast<int>

Reply:
That's #3181

Reply:
OK, again, sorry. Will go read it.